Sources/SotoCore/Credential/CredentialProvider.swift

```diff
@@ -107,10 +107,14 @@ extension CredentialProviderFactory {
         }
     }
 
-    /// Use this method to load credentials from your aws cli credential file, normally located at `~/.aws/credentials`
-    public static func configFile(credentialsFilePath: String = "~/.aws/credentials", profile: String? = nil) -> CredentialProviderFactory {
+    /// Use this method to load credentials from your AWS cli credentials and optional profile configuration files, normally located at `~/.aws/credentials` and `~/.aws/config`.
+    public static func configFile(
+        credentialsFilePath: String = ConfigFile.defaultCredentialsPath,
+        configFilePath: String = ConfigFile.defaultProfileConfigPath,
+        profile: String? = nil
+    ) -> CredentialProviderFactory {
         return Self { context in
```

Comment:
Use `RotatingCredentialProvider` instead of `DeferredCredentialProvider` here otherwise it will never check for expiring credentials from the sts assume credential provider

Reply:
👍